python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py

```diff
@@ -503,28 +503,32 @@ def _compute_ase_neighbors(atoms, options, dtype, device):
             cutoff=options.engine_cutoff(engine_length_unit="angstrom"),
         )
 
```

Comment:
Maybe add a small explanation about why the code is written this way for the next person coming and trying to improve readability by converting it back to a loop?